mlflow/pytorch/torchserve/serve.py

```diff
@@ -0,0 +1,128 @@
+
+
+class TorchServePlugin(BaseDeploymentClient):
+    def __init__(self, uri):
+        super(TorchServePlugin, self).__init__(target_uri=uri)
+        self.server_config = Config()
+        self.management_api = "http://localhost:8081"
+        self.inference_api = "http://localhost:8080"
+
+    def create_deployment(self, name, model_uri, flavor=None, config=None):
+        mar_file_path = self.generate_mar_file(
+            model_name=name,
+            version=self.server_config["version"],
+            model_file=self.server_config["model_file"],
+            handler_file=self.server_config["handler_file"],
+            model_uri=model_uri,
+        )
+
+        return self.register_model(mar_file_path=mar_file_path)
+
+    def update_deployment(self, name, model_uri=None, flavor=None, config=None):
+        url = "{}/{}/{}?{}".format(self.management_api, "models", name, "min_worker=3")
+        resp = requests.put(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return resp.text
+
+    def delete_deployment(self, name):
+        version = self.server_config["version"]
+        url = "{}/{}/{}/{}".format(self.management_api, "models", name, version)
+        resp = requests.delete(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return resp.text
+
+    def list_deployments(self):
+        url = "{}/{}".format(self.management_api, "models")
+        resp = requests.get(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return resp.text
+
+    def get_deployment(self, name):
+        url = "{}/{}/{}".format(self.management_api, "models", name)
+        resp = requests.get(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return resp.text
+
+    def predict(self, deployment_name, data):
+        version = self.server_config["version"]
+        url = "{}/{}/{}/{}".format(
+            self.inference_api, "predictions", deployment_name, version
+        )
+        data = {"data": data}
+        resp = requests.post(url, data)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return resp.text
+
+    def generate_mar_file(
+        self, model_name, version, model_file, handler_file, model_uri
+    ):
+        export_path = self.server_config["export_path"]
+        if export_path:
+            model_store = export_path
+        else:
+            model_store = "model_store"
+            if not os.path.isdir(model_store):
+                os.makedirs(model_store)
+
+        cmd = (
+            "torch-model-archiver --force --model-name {} "
+            "--version {} --model-file {} --serialized-file {} "
+            "--handler {} --export-path {}".format(
+                model_name, version, model_file, model_uri, handler_file, model_store
```

Comment:
We need to handle packaging of the extra files as well - eg the vocab file for text classification models. Please see the extra files param for torch-model-archiver https://github.com/pytorch/serve/tree/master/model-archiver#arguments